File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    index = 0
    answer = []
    
    while index < len(progresses):
        for i in range(index, len(progresses)):
            progresses[i] += speeds[i]
        
        new_index = index
        while new_index < len(progresses) and progresses[new_index] >= 100:
            new_index += 1
         

        if new_index != index:
            answer.append(new_index - index)
        index = new_index
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] days = new int[speeds.length];
        
        for (int i = 0; i < speeds.length; i++) {
            days[i] = (int) Math.ceil((100.0 - progresses[i]) / speeds[i]);
        }
        
        int i = 0;
        List<Integer> answer = new ArrayList<>();
        while (i < days.length) {
            int j = i + 1;
            
            while (j < days.length && days[i] >= days[j]) {
                j++;
            }
            answer.add(j - i);
            i = j;
        }
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}
